docs/extensibility/resources.md

```diff
@@ -97,6 +101,48 @@ If you target elements of an array rather that the array itself, you can use `it
 ]
 ```
 
+### Variable fields
+
+Additionally it's possible to define variable fields. In this case **path** is ommited and instead a **var** argument is used to specify variable name to assign to. Such a value is not added to the resultant yaml, but instead stored in memory and provided to any [JSONata](https://docs.jsonata.org/overview.html) handlers as variables (e.g. `$foo`). Variables will be provided for the current context - so if a variable is defined inside an array, the value is for that specific item. To access raw values, the predefined `$vars` variable has to be used.
+
+When using a variable inside an array it has to be wrapped inside a `[]` element (see example).
+
+### Item parameters
+
+- **var** - _[required]_ variable name.
+- **type** - _[required]_ type of field, as defined by JSON Schema.
+
+All other fields can be used analogous to regular form items (except for the **path** and **children** parameters).
+
+### Predefined variables
+
+All jsonata expressions have a few variables that are predefined instead of being read from variable fields.
+
+- **\$item** - when in an array this contains the current item.
+- **\$index** - contains the index of the current item within the array.
+- **\$indexes** - contains a list of all the indexex for nested arrays.
+- **\$vars** - contains a map of all the raw variable values.
+
+### Example
+
+In the example the visibility for item price and color are analogous - the former one uses scoped variables for the current item, the latter extracts the value from an array variable using provided index - this is mostly useful for complex scenarios only.
+
+```json
+[
+  { "var": "useDescription", "type": "boolean" },
+  { "path": "spec.description": "visibility": "$useDescription" },
+  { "path": "spec.items", "widget": "GenericList": "children": [
+    { "path": "[]", "children": [
+      { "path": "name" },
+      { "var": "itemMode", "type": "string", "enum": ["simple", "verbose"] },
+      { "path": "price", "visibility": "$itemMode = 'verbose'" },
```

Comment:
This notation doesn't seem to work when we have 2+ arrays with the same selector i.e. we have the same JSON in :

1. trafficPolicy (works)
2. port -> trafficPolicy (if trafficPolicy from 1st point is not selected, switches value for point 2. and 3., otherwise doesn't work)
3. subset -> trafficPolicy (if trafficPolicy from 1st point is not selected, switches value for point 2. and 3., otherwise doesn't work)
4. subset -> port -> works as long 1 or 2 or 3 are not selected
```
       {
                        "var": "consistentHashSelector",
                        "type": "string",
                        "enum": ["simple", "httpHeaderName"]
                      },
                      {
                        "path": "httpHeaderName",
                        "visibility": "$consistentHashSelector = 'httpHeaderName'"
                      },
```

Reply:
variable names are not scoped. they have to be unique across the config. i didn't mention it in the docs?